src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -1652,5 +1652,64 @@ public void ChainWith10DataAvailableHeaders_PartialValidationSucceededOnTheFirst
             listOfHeaders.Should().HaveCount(1);
             listOfHeaders.First().ChainedHeader.HashBlock.Should().Be(secondHeader.HashBlock);
         }
+
+        /// <summary>
+        /// Issue 40 @ CT advances after last checkpoint to height LC + MaxReorg + 10. New chain is presented with
+        /// fork point at LC + 5, chain is not accepted.
+        /// </summary>
+        [Fact]
+        public void ChainAdvancesAfterACheckpoint_NewChainIsPresentedWithForkPoint_ChainIsNotAccepted()
+        {
+            // Chain header tree setup. Initial chain has 5 headers and checkpoint at h5.
+            // Example: h1=h2=h3=h4=(h5).
+            const int initialChainSize = 5;
+            TestContext ctx = new TestContextBuilder().WithInitialChain(initialChainSize).UseCheckpoints().Build();
+            ChainedHeaderTree cht = ctx.ChainedHeaderTree;
+            ChainedHeader chainTip = ctx.InitialChainTip;
+
+            const int checkpointHeight = 5;
+            var checkpoint = new CheckpointFixture(checkpointHeight, chainTip.Header);
+            ctx.SetupCheckpoints(checkpoint);
+
+            // Setup max reorg to 100.
+            const int maxReorg = 100;
+            ctx.ChainStateMock.Setup(x => x.MaxReorgLength).Returns(maxReorg);
+
+            // Extend the chain with (checkpoint + MaxReorg + 10) headers, i.e. 115 headers.
+            const int extensionSize = 10;
+            const int chainASize = checkpointHeight + maxReorg + extensionSize;
+            ChainedHeader chainATip = ctx.ExtendAChain(chainASize, chainTip);
+
+            // Chain A is presented by peer 1.
+            List<BlockHeader> listOfChainABlockHeaders = ctx.ChainedHeaderToList(chainATip, chainASize);
+            ChainedHeader consumed = cht.ConnectNewHeaders(1, listOfChainABlockHeaders).Consumed;
+            ChainedHeader[] consumedChainAHeaders = consumed.ToArray(chainASize);
+            ChainedHeader[] originalChainAHeaders = chainATip.ToArray(chainASize);
+
+            // Sync all blocks from chain A.
+            for (int i = 0; i < chainASize; i++)
+            {
+                ChainedHeader currentChainTip = consumedChainAHeaders[i];
+                Block block = originalChainAHeaders[i].Block;
+
+                cht.BlockDataDownloaded(currentChainTip, block);
+                cht.PartialValidationSucceeded(currentChainTip, out bool fullValidationRequired);
+                ctx.FinalizedBlockMock.Setup(m => m.GetFinalizedBlockHeight()).Returns(currentChainTip.Height - maxReorg);
+                cht.ConsensusTipChanged(currentChainTip);
+            }
+
+            // Create new chain B with 20 headers and a fork point at height 10.
+            const int forkPointHeight = 10;
+            const int chainBSize = 20;
+            ChainedHeader forkTip = chainATip.GetAncestor(checkpointHeight + forkPointHeight);
+            ChainedHeader chainBTip = ctx.ExtendAChain(initialChainSize + chainBSize - forkPointHeight, forkTip);
```

Comment:
quite confusing this

imho

fpH is 10
forkTip should be GA(fpH) only
cBTip should be EC(cBSize - fpH)